symphony-bdk-core/src/main/java/com/symphony/bdk/core/activity/AbstractActivity.java

```diff
@@ -97,17 +103,27 @@ protected void processEvent(V4Initiator initiator, E event) throws EventExceptio
     // executes matcher with no failure
     final Optional<Boolean> matcherResult = this.executeMatcher(context);
     if (matcherResult.isPresent() && Boolean.TRUE.equals(matcherResult.get())) {
-      try {
-        log.trace("Before activity execution");
-        this.onActivity(context);
-      } catch (EventException ex) {
-        throw ex; // to allow events to be re-queued in DFv2 loop
-      } catch (Exception ex) {
-        log.warn("Activity execution failed.", ex);
+      if (isAsynchronous()) {
+        ExecutorService executorService = Executors.newFixedThreadPool(2);
```

Comment:
This executor should be initialised once at startup time, because it is an expensive operation, and the threads in the pool is supposed to be reused when needed. 
Here about the thread pool, maybe we can start with newCachedThreadPool to have unbounded pool.

Reply:
also we should use a custom threadFactory for this thread pool, so that we can inherit the traceId from the parent thread to the new thread.